HerPublicWebsite.BusinessLogic/Services/RegularJobs/PolicyTeamUpdateService.cs

```diff
@@ -15,21 +15,25 @@ public class PolicyTeamUpdateService : IPolicyTeamUpdate
     private readonly ICsvFileCreator csvFileCreator;
     private readonly IWorkingDayHelperService workingDayHelperService;
     private readonly IEmailSender emailSender;
+    private readonly IReferralFilterService referralFilterService;
     
     public PolicyTeamUpdateService(
         IDataAccessProvider dataProvider,
-        ICsvFileCreator csvFileCreator, 
+        ICsvFileCreator csvFileCreator,
         IWorkingDayHelperService workingDayHelperService,
-        IEmailSender emailSender
+        IEmailSender emailSender,
+        IReferralFilterService referralFilterService
     )
     {
         this.dataProvider = dataProvider;
         this.csvFileCreator = csvFileCreator;
         this.workingDayHelperService = workingDayHelperService;
         this.emailSender = emailSender;
+        this.referralFilterService = referralFilterService;
     }
 
-    public async Task SendPolicyTeamUpdate(){
+    public async Task SendPolicyTeamUpdate()
```

Comment:
I'm kind of in two minds here and interested in your opinion. I think the useful thing to have a test covering "when we construct the policy team update, we only end up looking at referrals not submitted to pending LAs". As a unit test for this service, that probably looks like `given our mocked DataAccessProvider returns these referrals` -> `then we only call the CsvFileCreator methods with the ones that don't have that flag`.

Do you think there's value in that?

Reply:
I think where I land is that there's now some non-trivial business logic in this service, so we should test it, even though it's mostly just going to be testing that one thing is wired up to another. In an ideal world we'd have integration tests, but that's more effort than it's worth now.

Reply:
have added two tests for the two services. since all input services are mocked the filter logic is mocked out too, but the tests do verify that the results of the filter are respected in calls to csv creator / email creator